src/config/read.rs

```diff
@@ -0,0 +1,47 @@
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+
+use crate::config::{Config, Source};
+
+pub struct ConfigReader {
+  config: Config,
+}
+
+impl ConfigReader {
+  pub fn init() -> Self {
+    Self { config: Config::default() }
+  }
+  pub async fn serialize_config(&mut self, path: &String) -> anyhow::Result<()> {
+    let conf = if path.starts_with("http://") || path.starts_with("https://") {
```

Comment:
we could just try parse this to [Url](https://docs.rs/reqwest/0.11.22/reqwest/struct.Url.html) and not check the schema manually and actually be able to use any other protocol that is supported by the reqwest

Reply:
The goal of this pr was to add fetch over HTTP

the code would be very different for fetching with… say websocket, can we discuss this on Discord?

Reply:
the main goal that I wanted to achieve with my comment is to simplify the code and make it easier to extend in the future if we need to. For now we a targeting only http as you rightly pointed out.

tag me on discord if you want to discuss anything further